python-checks/src/main/java/org/sonar/python/checks/hotspots/AbstractCookieFlagCheck.java

```diff
@@ -0,0 +1,155 @@
+        if (secureArgument == null || isFalsy(secureArgument.expression())) {
+          ctx.addIssue(callExpression.callee(), message());
+        }
+      }
+    });
+  }
+
+  @CheckForNull
+  private static RegularArgument getArgument(List<Argument> arguments, int nArg, String argumentName) {
+    return arguments.stream()
+      // in call site of this function, argument will always be of type RegularArgument because we check no argument is unpacking
+      .map(RegularArgument.class::cast)
+      .filter(argument -> {
+        Name keywordArgument = argument.keywordArgument();
+        return keywordArgument != null && keywordArgument.name().equals(argumentName);
+      })
+      .findFirst().orElseGet(() -> checkSensitivePositionalArgument(arguments, nArg));
+  }
+
+  private static RegularArgument checkSensitivePositionalArgument(List<Argument> arguments, int nArg) {
+    for (int i = 0; i < arguments.size(); i++) {
+      // in call site of this function, argument will always be of type RegularArgument because we check no argument is unpacking
+      RegularArgument arg = (RegularArgument) arguments.get(i);
+      if (i == nArg) {
+        return arg;
+      }
+      if (arg.keywordArgument() != null) {
+        // not positional anymore
+        return null;
+      }
+    }
+    return null;
+  }
+
+  private static Stream<SubscriptionExpression> getSubscriptionToCookies(List<ExpressionList> lhsExpressions) {
+    return lhsExpressions.stream()
+      .flatMap(expressionList -> expressionList.expressions().stream())
+      .filter(lhs -> {
+        if (lhs.is(Kind.SUBSCRIPTION)) {
+          SubscriptionExpression sub = (SubscriptionExpression) lhs;
+          Symbol objectSymbol = getObjectSymbol(sub.object());
+          return "http.cookies.SimpleCookie".equals(SymbolUtils.getTypeName(objectSymbol));
+        }
+        return false;
+      })
+      .map(SubscriptionExpression.class::cast);
+  }
+
+  private static boolean isSettingFlag(SubscriptionExpression sub, String flagName) {
+    List<ExpressionList> subscripts = getSubscripts(sub);
+    if (subscripts.size() == 1) {
+      return false;
+    }
+    return subscripts.stream()
+      .skip(1)
+      .anyMatch(s -> s.expressions().size() == 1 && isStringLiteral(s.expressions().get(0), flagName));
+  }
+
+  private static List<ExpressionList> getSubscripts(SubscriptionExpression sub) {
+    Deque<ExpressionList> subscripts = new ArrayDeque<>();
+    subscripts.addFirst(sub.subscripts());
+    Expression object = sub.object();
+    while (object.is(Kind.SUBSCRIPTION)) {
+      subscripts.addFirst(((SubscriptionExpression) object).subscripts());
+      object = ((SubscriptionExpression) object).object();
+    }
+    return new ArrayList<>(subscripts);
+  }
+
+  private static boolean isStringLiteral(Expression expression, String value) {
```

Comment:
Maybe we can have a more explicit name for this method? Like `isFlagNameStringLiteral` or something in that fashion?